pkg/skaffold/runner/v1/dev.go

```diff
@@ -170,10 +168,8 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *logge
 			r.intents.ResetDeploy()
 		}()
 
-		for _, w := range watchers {
-			logrus.Debugf("stopping watcher %s", w.Name())
-			w.Stop()
-		}
+		r.deployer.GetResourcePreviewer().Stop()
```

Comment:
I wonder if `GetResourcePreviewer().Stop()`, `GetDebugger().Stop()`, etc. can be called from the individual deployers directly instead of in `dev.go` or the runners. Then we wouldn't need the `DebuggerMux`, `ResourcePreviewerMux`, etc. Also since these elements are already decomposed in this PR to be independent for each constituent `Deployer`, it seems counter intuitive to again accumulate them in `..Mux` and calling them together.

Reply:
https://github.com/GoogleContainerTools/skaffold/pull/6021#issuecomment-861869621